pkg/openapi/crdSyncNew.go

```diff
@@ -0,0 +1,139 @@
+package openapi
+
+import (
+	"fmt"
+	"time"
+
+	prun "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	client "github.com/nirmata/kyverno/pkg/dclient"
+	crdv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	crdInformer "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions/apiextensions/v1"
+	crdLister "k8s.io/apiextensions-apiserver/pkg/client/listers/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type crdSyncNew struct {
+	client          *client.Client
+	controller      *Controller
+	crdLister       crdLister.CustomResourceDefinitionLister
+	crdListerSynced cache.InformerSynced
+	queue           workqueue.RateLimitingInterface
+}
+
+func NewCRDSyncNew(client *client.Client, controller *Controller, crdInformer crdInformer.CustomResourceDefinitionInformer) *crdSyncNew {
+	if controller == nil {
+		panic(fmt.Errorf("nil controller sent into crd sync"))
+	}
+
+	cs := &crdSyncNew{
+		controller: controller,
+		client:     client,
+		queue:      workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "crds"),
+	}
+
+	crdInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			if err == nil {
+				cs.queue.Add(key)
+			}
+		},
+		UpdateFunc: func(old interface{}, new interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(new)
+			if err == nil {
+				cs.queue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			cs.controller.deleteCRDFromCache(obj.(*crdv1.CustomResourceDefinition).Spec.Names.Kind)
+		},
+	})
+
+	cs.crdLister = crdInformer.Lister()
+	cs.crdListerSynced = crdInformer.Informer().HasSynced
+	stopCh := make(chan struct{})
```

Comment:
Can we use the same stopCh from main? Otherwise how is this stopCh been controlled?